net/vsftpd/files/vsftpd.init

```diff
@@ -2,12 +2,151 @@
 # Copyright (C) 2006-2011 OpenWrt.org
 
 START=50
+USE_PROCD=1
+BIN=vsftpd
 
-start() {
-	mkdir -m 0755 -p /var/run/vsftpd
-	service_start /usr/sbin/vsftpd
+. /lib/functions.sh
+
+PORT=21
+OUTPUT_CONF="/var/etc/vsftpd.conf"
+readonly TEMP_OUTPUT_CONF="/var/etc/vsftpd.conf.tmp"
+
+write_conf() {
+	local key="$1"
+	local value="$2"
+
+	if [ -n "$key" ] && [ -n "$value" ]; then
+		echo "$key=$value" >> "$TEMP_OUTPUT_CONF"
+	fi
+}
+
+write_conf_bool() {
+	local key="$1"
+	local value="$2"
+
+	if  [ "$value" = "1" ]; then
+		write_conf "$key" "YES"
+	else
+		write_conf "$key" "NO"
+	fi
+}
+
+validate_vsftpd_section() {
+	uci_load_validate vsftpd global "$1" "$2" \
+		'background:bool:0' \
```

Comment:
[If I remember correctly](https://openwrt.org/docs/guide-developer/procd-init-scripts#procd_init_scripts), then the job must not be set to background If `procd` is used. The service must run in foreground.

So I think the option `background` should not be configurable via uci . Please always set the option `background=NO` to the generated `vsftpd` config.